lib/game_teams_stats.rb

```diff
@@ -28,17 +26,15 @@ def total_goals_by_team_id(team_id)
   end
 
   def average_goals_per_team(team_id)
-    total_goals_by_team_id(team_id) / total_games_by_team_id(team_id)
+    total_goals_by_team_id(team_id).to_f / total_games_by_team_id(team_id).to_f
```

Comment:
Only one of these has to be a float. No need to worry though. The module I'm working on will incorporate float within the generic average method.